src/escape.d

```diff
@@ -299,9 +305,25 @@ bool checkEscapeRef(Scope* sc, Expression e, bool gag)
                     }
                     else if (sc._module && sc._module.isRoot())
                     {
-                        //printf("escaping reference to local ref variable %s\n", v.toChars());
-                        //printf("storage class = x%llx\n", v.storage_class);
-                        error(loc, "escaping reference to local ref variable %s", v);
+                        Dsymbol p = v.toParent2();
+                        if (p == sc.func)
+                        {
+                            //printf("escaping reference to local ref variable %s\n", v.toChars());
+                            //printf("storage class = x%llx\n", v.storage_class);
+                            error(loc, "escaping reference to local ref variable %s", v);
+                            return;
+                        }
+                        // Don't need to be concerned if v's parent does not return a ref
+                        if (FuncDeclaration fd = p.isFuncDeclaration())
+                        {
+                            if (fd.type && fd.type.ty == Tfunction)
```

Comment:
```d
if (FuncDeclaration fd = p.isFuncDeclaration() &&
        fd.type && fd.type.ty == Tfunction)
{
    ...
}
```

Your code (even in this PR) does use multiline Boolean conditions, so this variation is redundant and jarring.